VikingQuiz.Api/Controllers/EmailController.cs

```diff
@@ -1,63 +1,128 @@
-            })
+            using (MailMessage message = new MailMessage(FromAddress, receiverAddress) { Subject = subject, Body = body })
             {
                 client.Send(message);
             }
         }
 
+        [Route("email")]
+        [HttpGet("{id:int}")]
+        [AllowAnonymous]
+        public IActionResult SendEmail(int id)
+        {
+            User user = userRepository.GetUserById(id);
+
+            if (user == null)
+            {
+                return NotFound();
+            }
+
+            string Subject = "Register your account";
+            string token = authenticationService.GenerateTokenForUser(user, ConfirmEmailExpiryHours, "email");
+            string registerAccountAddress = frontendBaseUrl + "/confirm/" + token;
+            string Body = "Click the following link to register your account\n\n" + registerAccountAddress;
+            this.SendEmail(user.Email, Subject, Body);
+            return Ok();
+        }
+
         [Route("token")]
-        [HttpGet("{token}")]
-        public string ValidateAddress(string token)
+        [Authorize(Roles = "email")]
+        public string ActivateAccount()
         {
-            userRepository.Activate(token);
+            int userId = User.Claims.GetUserId();
+            userRepository.Activate(userId);
             return "validation worked";
         }
+
+        [HttpPost]
+        [AllowAnonymous]
+        public IActionResult SendPasswordResetEmail([FromBody] EmailViewModel email)
+        {
+            User user = userRepository.GetUserByEmail(email.Email);
+
+            if (user == null)
+            {
+                return NotFound();
+            }
+
+            string Subject = "Reset your password";
+            string token = authenticationService.GenerateTokenForUser(user, ResetPasswordExpiryHours, "ResetPassword");
+            string resetPasswordAddress = frontendBaseUrl + "/forgot/" + token;
+            string Body = "Click the following link to reset your password\n\n" + resetPasswordAddress +
+                "\n\nIf you didn't request this password reset, ignore this message";
+
+            this.SendEmail(user.Email, Subject, Body);
+            return Ok();
+        }
+
+        [HttpPut]
+        [Authorize(Roles = "ResetPassword")]
+        public IActionResult ResetPassword([FromBody] PasswordViewModel password)
+        {
+            string newPassword = password.Password.SHA256Encrypt();
+
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState.GetErrors().First());
+            }
+
+            int userId = User.Claims.GetUserId();
+            User user = userRepository.GetUserById(userId);
+
+            if (user == null)
+            {
+                return NotFound();
+            }
+
+            if (user.Pass == newPassword)
+            {
+                return BadRequest("your new password cannot be the same as the old one");
```

Comment:
If this is a message that is directly displayed, make it look good here. Capital first letter, punctuation.
I this message is not displayed, return a code and treat it in the client.